frontend/javascripts/admin/onboarding.js

```diff
@@ -475,14 +476,15 @@ class OnboardingView extends React.PureComponent<Props, State> {
   }
 
   render() {
+    const skipAddingDatasets = features().skipAddingDatasetsInOnboarding;
     const currentStepContent = (() => {
       switch (this.state.currentStep) {
         case 0:
           return this.renderCreateOrganization();
         case 1:
           return this.renderCreateAccount();
         case 2:
-          return this.renderUploadDatasets();
+          return skipAddingDatasets ? this.renderWhatsNext() : this.renderUploadDatasets();
```

Comment:
I'd suggest to have a method like this:

```
getAvailableSteps() {
  if (demoInstance) {
  	return [
		["Create Organization", this.renderCreateOrganization],
		["Create Account", this.renderCreateAccount],
		["What's Next?", this.renderWhatsNext],
  	]
  } else {
  	return [
		["Create Organization", this.renderCreateOrganization],
		["Create Account", this.renderCreateAccount],
		["Add Dataset", this.renderUploadDatasets],
		["What's Next?", this.renderWhatsNext],
  	]
  }
}
```

Then, this method can be used for `currentStepContent` and when defining the children of `<Step>`. For that, the render methods need to be changed to `render... = () => {}` to fix the binding.

Reply:
Nice suggestion! I took it a bit further and used objects inside the array with the structure 
```{ title:"...", component: some_function }```.
 I think this makes the code that uses this function's returned array a bit easier to understand :slightly_smiling_face:

Reply:
Awesome! :)

Reply:
> For that, the render methods need to be changed to `render... = () => {}` to fix the binding.

@MichaelBuessemeyer I fixed this in #4067 now :) Please make sure to always test your changes, especially when you change something after the PR was already approved (and thus tested by the reviewer).

Reply:
Ok sorry, looks like I forgot that 🙈.